File: step2/src/movies/main.cpp
#include "Interpreter.h"
#include <fstream>
#include <iostream>

int main() {
    Interpreter interpreter;

    interpreter.ImportFile();

    std::cout<<"Please, indicate a decade ("<< "1900" <<"/"<< "2010" << ")";

    int decade;;
    while(true){
        std::cin>>decade;
        if(std::cin.fail()){
            std::cin.clear();
            std::cin.ignore();
            std::cout<<"Only numbers and decades, please"<< std::endl;
        } else {
            decade = interpreter.InterpretDecade(decade);
            break;
        }
    }

    std::vector<Movie> filteredMovies = interpreter.FilterMovies(decade);
    interpreter.ExportFile(decade, filteredMovies);

  return 0;
}

File: step2/src/movies/main.py

from Interpreter import Interpreter


if __name__ == "__main__":
    Interpreter.ImportFile()
    
    while True:
        try:
            decade = int(input('Type a decade 10-20...80 \n'))
            break
        except:
            print("That's not a valid option")

    decade = Interpreter.InterpretDecade(decade)
    filteredMovies = Interpreter.FilterMovies(decade)
    Interpreter.ExportFile(decade, filteredMovies)
